src/App.js

```diff
@@ -23,46 +23,80 @@ const generateSquares = () => {
   }
 
   return squares;
-}
+};
 
 const App = () => {
 
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(true);
+  const [winner, setWinner] = useState('');
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const onClickCallback = (updatedSquare) => {
+    let updatedBoard = [];
+    for (let row = 0; row < squares.length; row++) {
+      for (let col = 0; col < squares.length; col++) {
+        if (updatedSquare === squares[row][col].id && squares[row][col].value === '') {
+          squares[row][col]['value'] = (swapPlayer() ? PLAYER_1 : PLAYER_2);
+        }
+      }
+      updatedBoard.push(squares[row]);
+    }
+    setSquares(updatedBoard);
+    checkForWinner();
+  };
 
+  function swapPlayer() {
+    setPlayer(!player);
+    return player;
+  };
 
   const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if 
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if 
-    //    all three squares have the same value.
+    // if the winner is already delcared, return (don't assign new winner)
+    if (winner !== '') return;
+    
+    // flatten the array of squares
+    const flatArray = [].concat(...squares);
+    // have a 2d array of combos to check
+    const winnerCombos = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
```

Comment:
This is clever! You might also consider how you could loop through the 2D array to find rows or columns of matches.